### ISSUE TITLE
Unhandled Promise Rejection: Error: abort(Error: WebAssembly.Module doesn't parse at byte xxx

### ISSUE THREAD

**Issue by Dewyzee** [opencv/opencv#19553]
- OpenCV => 4.5.1
- Operating System / Platform =>Mac OS 10.15.4
- Compiler => Emscripten SDK

##### Detailed description

The result(opencv.js) compiled by Emscripten can be running well in chrome but goes wrong in Safari. Just as follows:
![image](https://user-images.githubusercontent.com/4916726/108323599-d3dde300-7201-11eb-8638-cdf8ddb7ca21.png)

##### Steps to reproduce

* clone opencv from github
* create branch from tag 4.5.1
* compile opencv use Emscripten
```bash
python ./platforms/js/build_js.py build_js --emscripten_dir ../emsdk/upstream/emscripten --build_flags="-s ASSERTIONS=1"
```
* get the result and move into my project, start the project and open it in safari

**Comment by LuanBaviloni** [comment]
Any news on this? I'm getting the same error.

**Comment by LuanBaviloni** [comment]
I'm getting the same error.

**Comment by LuanBaviloni** [comment]
I think I've figured out the reason for this.

I tried to compile OpenCV 3.4.13 using one of the latest version of Emscripten SDK (version 2.0.16), but this version is generating a **.wasm** file with a **DataCount** section, that shouldn't be there if you're compiling without threads optimization. It has to do with **Bulk Memory Operations**, a feature not supported yet in Safari browser.

I'm not an WebAssembly expert, but I believe that it might be a bug in the newest Emscripten SDK versions.

I was able to compile the OpenCV using 2.0.10 version of the SDK:

`emsdk update`
`emsdk install 2.0.10`
`emsdk activate 2.0.10`

Read more at:
https://github.com/WebAssembly/bulk-memory-operations/blob/master/proposals/bulk-memory-operations/Overview.md
https://webassembly.org/roadmap/

**Comment by jeroenvantilburg** [comment]
I had the same problem. However, going back to Emscripten 2.0.10 did not make any difference. My own build of OpenCV.js did not work on Safari, but it did work on FireFox and Chrome. But when I go back to Emscripten 1.39.0, the problem in Safari went away:
`emsdk install 1.39.0` 
`emsdk activate 1.39.0` 
`source ./emsdk_env.sh`
`cd /Applications/opencv-4.5.2/`
`python3 platforms/js/build_js.py build_wasm --build_wasm --emscripten_dir /Applications/emsdk/upstream/emscripten`

It took me two days to find this out. LuanBaviloni's suggestion to go back to an older version of Emscripten helped me on the right track. For FF and Chrome with the emscripten 2.0.18 build, I found out that I had to resolve the Promise before I could access the cv functions, as done for instance in the [tests.html](https://github.com/opencv/opencv/blob/master/modules/js/test/tests.html#L50-L62). Interestingly the standard build in https://docs.opencv.org/4.5.2/opencv.js does not return a Promise and when inspecting this file there seems to be no use of Promises. This got me to check the nightly builds on Opencv.org, since all I really wanted is to have the same stable build as that one. It seems that Emscripten 1.39.0 is used to build opencv.js. With this version I can finally build my own stable opencv.js.

One thing I still do not understand is that in the nightlies the default build target is used (should be asm?) instead of the --build_wasm option. On top of that there seems to be hardly any byte difference in the resulting opencv.js files whether or not having this option.
